telluric/util/raster_utils.py

```diff
@@ -65,22 +65,45 @@ def _get_telluric_tags(source_file):
         return return_tags
 
 
-def convert_to_cog(source_file, destination_file, resampling=Resampling.gauss):
+def _create_options_for_cog(create_options, source_profile, blocksize):
+    """
+    it uses the profile of the source raster, override anything using the creation_options
+    and grantees we will have tiled raster and blocksize
+    """
+    if not(create_options):
+        create_options = {}
+
+    create_options["blocksize"] = blocksize
+    create_options["tiled"] = True
+    for key in ["nodata", "compress"]:
+        if key not in create_options:
+            create_options[key] = source_profile.get(key, None)
+    return create_options
+
+
+def convert_to_cog(source_file, destination_file, resampling=rasterio.enums.Resampling.gauss, blocksize=256,
+                   overview_blocksize=256, create_options=None):
     """Convert source file to a Cloud Optimized GeoTiff new file.
 
     :param source_file: path to the original raster
     :param destination_file: path to the new raster
     :param resampling: which Resampling to use on reading, default Resampling.gauss
+    :param blocksize: the size of the blocks default 256
+    :param overview_blocksize: the block size of the overviews, default 256
+    :param create_options: <dictioanry>, options that can override the source raster profile,
+                          notice that you can't override tiled=True, and the blocksize
```

Comment:
The user can't override the `blocksize` because it's a separate parameter of the function, right?

Reply:
yes